src/interfaces/create-client-options.interface.ts

```diff
@@ -9,6 +10,9 @@ export interface CreateClientOptions {
   onRedirectCallback?: (_: RedirectParams) => void;
   onBeforeAutoRefresh?: () => boolean;
   onRefresh?: (_: AuthenticationResponse) => void;
+  onRefreshFailure?: (args: {
+    signIn: Awaited<ReturnType<typeof createClient>>["signIn"];
```

Comment:
i somewhat hate that the client isn't a class and i can't just pass the whole instance into this callback.

(OTOH, getting the `signIn` function directly like this may feel more familiar to authkit-react users who never deal with the client directly).

We may want to add other stuff to this callback in the future (`organizationId`? `user` ?). I think I'll leave it bare-bones for now though.

Not totally sold on the name? `onRefreshFailure` vs `onSessionEnded`? (`onRefreshError` seems more generally descriptive/true of why this thing is being called).